File: week-04/day-01/Counter/counter.py
class Counter(object):
    def __init__(self, number = 0):
        self.count_number = number

    def add(self, number = None):
        if number != None:
            self.count_number += number
        else:
            self.count_number +=1

    def get(self):
        return self.count_number
    
    def reset(self):
        self.__init__()

File: Csharp-course/week-03/day-03/CounterProject/CounterProject/Counter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CounterProject
{
    class Counter
    {
        int prop;

        public Counter(int num = 0)
        {
            this.prop = num;
        }

        public void Add(int number = 0)
        {
            if (number > 0)
            {
                this.prop += number;
            }
            else
            {
                this.prop++;
            }
        }

        public string Get()
        {
            return this.prop.ToString();
        }

        public void Reset()
        {
            this.prop = 0;
        }
    }
}
